exercises/bracket-push/example.ml

```diff
@@ -0,0 +1,26 @@
+open Core.Std
+
+type 'a stack = 'a list
+
+let push (s: 'a stack) (a: 'a): 'a stack = a :: s
+let pop (s: 'a stack): ('a * 'a stack) option = match s with
+  | [] -> None
+  | x :: xs -> Some (x, xs)
+
+let foldThroughOptionMonad (xs: 'a list) (z: 'b) ~(f:'b -> 'a -> 'b option): 'b option =
```

Comment:
Thanks very much for your feedback. You make a point that I hadn't considered around the example implementation file, and I think it is a fair one.

It may be as well to consider anyone visiting the repo for hints (I've done the same myself). Monads in Ocaml are not as all-pervasive as they are in Haskell, so they may make the code look intimidating, which I certainly don't want.

I thought the best bet was to specialise the implementation to the problem in hand, rather than write a monadic fold function. I also added some documentation, which I hope might help.